draft-ietf-masque-h3-datagram.md

```diff
@@ -249,23 +141,30 @@ corresponding stream.
 # Capsules {#capsule}
 
 This specification introduces the Capsule Protocol. The Capsule Protocol is a
-sequence of type-length-value tuples that allows endpoints to reliably
-communicate request-related information end-to-end, even in the presence of
-HTTP intermediaries.
-
-
-## Capsule Protocol {#capsule-protocol}
+sequence of type-length-value tuples that new HTTP Methods or new HTTP Upgrade
+Tokens can choose to use. It allows endpoints to reliably communicate
+request-related information end-to-end on HTTP request streams, even in the
+presence of HTTP intermediaries. The Capsule Protocol can be used to exchange
+HTTP Datagrams when HTTP is running over a transport that does not support the
+QUIC DATAGRAM frame.
 
 This specification defines the "data stream" of an HTTP request as the
 bidirectional stream of bytes that follow the headers in both directions. In
 HTTP/1.x, the data stream consists of all bytes on the connection that follow
 the blank line that concludes either the request header section, or the 2xx
-(Successful) response header section. In HTTP/2 and HTTP/3, the data stream of
-a given HTTP request consists of all bytes sent in DATA frames with the
+(Successful) response header section. In HTTP/2 and HTTP/3, the data stream of a
+given HTTP request consists of all bytes sent in DATA frames with the
 corresponding stream ID. The concept of a data stream is particularly relevant
 for methods such as CONNECT where there is no HTTP message content after the
 headers.
 
+Note that use of the Capsule Protocol is not required to use HTTP Datagrams.
+Definitions of new HTTP Methods or of new HTTP Upgrade Tokens can use HTTP
+Datagrams with their own data stream protocol.
```

Comment:
This text feels like it could use some work. It seems like the situation is as follows:

1. You an use HTTP datagrams without capsules or any other encapsulation protocol if the transport supports QUIC datagram.
2. If the protocol doesn't support QUIC datagram you need some kind of encapsulation protocol. You are free to use capsules or define your own thing in this case.

Is this right?

Reply:
I think that's right, though I'd say that the spec nudges folks towards using capsules, while acknowledging that it's possible to bring your own encapsulation

Reply:
I wouldn't call the capsule protocol unnecessary, as it has value outside of datagrams: for example WebTransport uses it to send close details, and CONNECT-IP uses it to exchange configuration information.

Reply:
OK, well, feel free to revise my proposed text. However, the existing text needs revision.

Reply:
Done